src/enclave/rpchandler.h

```diff
@@ -23,8 +23,8 @@ namespace enclave
     virtual void tick(std::chrono::milliseconds elapsed_ms_count) {}
 
     // Used by rpcendpoint to process incoming client RPCs
-    virtual std::vector<uint8_t> process(
-      RPCContext& ctx, const std::vector<uint8_t>& input) = 0;
+    virtual std::optional<std::vector<uint8_t>> process(
```

Comment:
Rather than setting `ctx.is_pending`, this now indicates a pending transaction by returning `nullopt`. This lets us reason about the `RPCContext` as pure input (const).